sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -161,11 +162,12 @@ static pi_result USMSetIndirectAccess(pi_kernel kernel) {
 extern "C" {
 
 // Convenience macro makes source code search easier
-#define OCL(pi_api) Ocl##pi_api
+#define OCL(pi_api) pi_api
 
 // Example of a PI interface that does not map exactly to an OpenCL one.
-pi_result OCL(piPlatformsGet)(pi_uint32 num_entries, pi_platform *platforms,
-                              pi_uint32 *num_platforms) {
+__SYCL_EXPORT pi_result OCL(piPlatformsGet)(pi_uint32 num_entries,
```

Comment:
I'm not sure `__SYCL_EXPORT` is required in the source files. Header file changes should be enough. Right?

If not, pi_cuda.cpp must be updated as well.
NOTE: CUDA pre-commit validation will be enabled back tomorrow and I'd like to get the results for this check before merging.

Reply:
@bader , 
I believe you are right, but let me double check on my Windows machine.
I will work on this today, and update on this comment chain.

Reply:
@bader 
I double-checked on Windows & LInux, we only need to put __SYCL_EXPORT on pi.h.
We do not need to put __SYCL_EXPORT on pi**.cpp.
I confirmed that only pi* symbols are exported in libpi_opencl.so and pi_opencl.lib.
So, we do not need any changes in pi_cuda.cpp because of this change in pi.h.

Reply:
> I confirmed that only pi* symbols are exported 
Please add a test that will verify this.

Reply:
@smaslov-intel 
I added a regression test pi_opencl_symbol_check.cpp to check only the pi* symbols are exported.